services/web/server/src/simcore_service_webserver/conversations/_conversation_repository.py

```diff
@@ -0,0 +1,166 @@
+import logging
+from typing import cast
+
+from aiohttp import web
+from models_library.conversations import (
+    ConversationDB,
+    ConversationID,
+    ConversationPatchDB,
+    ConversationType,
+)
+from models_library.groups import GroupID
+from models_library.products import ProductName
+from models_library.projects import ProjectID
+from models_library.rest_ordering import OrderBy, OrderDirection
+from pydantic import NonNegativeInt
+from simcore_postgres_database.models.conversations import conversations
+from simcore_postgres_database.utils_repos import (
+    get_columns_from_db_model,
+    pass_or_acquire_connection,
+    transaction_context,
+)
+from sqlalchemy import asc, desc, func
+from sqlalchemy.ext.asyncio import AsyncConnection
+from sqlalchemy.sql import select
+
+from ..db.plugin import get_asyncpg_engine
+from .errors import ConversationErrorNotFoundError
+
+_logger = logging.getLogger(__name__)
+
+
+_SELECTION_ARGS = get_columns_from_db_model(conversations, ConversationDB)
+
+
+async def create(
+    app: web.Application,
+    connection: AsyncConnection | None = None,
+    *,
+    name: str,
+    project_uuid: ProjectID | None,
+    user_group_id: GroupID,
+    type_: ConversationType,
+    product_name: ProductName,
+) -> ConversationDB:
+    async with transaction_context(get_asyncpg_engine(app), connection) as conn:
+        result = await conn.execute(
+            conversations.insert()
+            .values(
+                name=name,
+                project_uuid=f"{project_uuid}" if project_uuid else None,
+                user_group_id=user_group_id,
+                type=type_,
+                created=func.now(),
+                modified=func.now(),
+                product_name=product_name,
+            )
+            .returning(*_SELECTION_ARGS)
+        )
+        row = result.one()
+        return ConversationDB.model_validate(row)
+
+
+async def list_project_conversations(
+    app: web.Application,
+    connection: AsyncConnection | None = None,
+    *,
+    project_uuid: ProjectID,
+    # pagination
+    offset: NonNegativeInt,
+    limit: NonNegativeInt,
+    # ordering
+    order_by: OrderBy,
```

Comment:
I still think we have to tune thisone... None of the schemas at the rest should be passed "as is" to the repo layer.